### ISSUE TITLE
Better naming of samples, for GI_, Geometry_

### ISSUE THREAD

**Issue by cryham** [OGRECave/ogre-next#472]
#### System Information
- Ogre Version: all

#### Detailled description
So I believe this **is** an issue. Also forum seems less direct.

I'm checking out GI methods recently. I have some understanding and some shorcuts in mind.
But the samples are IMO named in a confusing way.
I've read the nice docs page (/Docs/src/manual/GiMethods.md):
https://ogrecave.github.io/ogre-next/api/latest/_gi_methods.html
and older, all relevant links
https://www.ogre3d.org/2021/10/25/upcoming-global-illumination-improvements-in-ogre-next
https://www.ogre3d.org/2019/08/05/voxel-cone-tracing
https://www.ogre3d.org/2019/08/14/pcc-vct-hybrid-progress
https://www.ogre3d.org/2016/12/23/ogre-progress-report-december-2016

How is anyone supposed to know from filename which samples are showing GI methods, and which?
So far I got that:
```
Sample_ImageVoxelizer - is GI CIVCT
Sample_InstantRadiosity - has GI VPL or Irradiance Volumes, seems to be a term from 2009
Test_Voxelizer - has GI IFD and VCT - why is it a Test_ not Sample_ it has few more than Sample_ImageVoxelizer
LocalCubemaps, LocalCubemapsManualProbes, PccPerPixelGridPlacement - are for GI PCC
```

I see there are *only* such hints on docs page:
`- See Samples/2.0/Tests/Voxelizer`

So my point is, there isn't much info or at all in sample names.
I would *rename* at least to: `GI_InstantRadiosity, GI_ImageVoxelizer`, `GI_Voxelizer`, etc.
or if I got it right then even to: `GI_VPL_InstantRadiosity`, `GI_CIVCT_ImageVoxelizer`, `GI_IFD_Voxelizer`, `GI_PCC_LocalCubemaps`, etc.
And also the binaries accordingly, to: `Sample_*` and above names.

Because it is really confusing to see like 56 or 65 binary files (for a beginner user nor an expert) and understanding like half of those names.

I'm also suggesting another *group*, and renaming all samples related to geometry,
with e.g. `Geometry_` prefix, so far I see those would be:
`CustomRenderable, Decals?, DynamicGeometry, MeshLod, V2ManualObject, V2Mesh`

I think my idea is clear now, perhaps there could be other groups that would help too.
This is probably a lot to do and change in many places, and maybe too late?
BTW which GI would be good for SR3? Outdoor nature scenes (mostly) 1 to 2km square area, with lots of trees possible?

**Comment by darksylinc** [comment]
You're right. The naming is utterly confusing and I get lost too, which is why I wrote [this piece of documentation](https://ogrecave.github.io/ogre-next/api/latest/_gi_methods.html) but it's not enough.

> I would rename at least to: GI_InstantRadiosity, GI_ImageVoxelizer, GI_Voxelizer, etc.
or if I got it right then even to: GI_VPL_InstantRadiosity, GI_CIVCT_ImageVoxelizer, GI_IFD_Voxelizer, GI_PCC_LocalCubemaps, etc.

Sounds like a good idea! You got the name rights except for `GI_IFD_Voxelizer`

A couple remarks:

- Voxelizer contains more than IFD (it's VCT or IFD, it shows both. IFD depends on VCT to work).
- PCC is often considered a "reflections" algorithm rather than "GI", but others put it in the GI category (our [docs](https://ogrecave.github.io/ogre-next/api/latest/_gi_methods.html) put it in the GI)


I'm also suggesting another group, and renaming all samples related to geometry,
with e.g. Geometry_ prefix, so far I see those would be:
CustomRenderable, Decals?, DynamicGeometry, MeshLod, V2ManualObject, V2Mesh.

That.... sounds a good idea! (Decals does not belong to Geometry, but I agree for a beginner it sounds like it should)

**PRs are welcome!**

I suggest to do the PR in batches. For example put all the Geometry in one folder; one PR.
Another for the GI.

> BTW which GI would be good for SR3? Outdoor nature scenes (mostly) 1 to 2km square area, with lots of trees possible?

Outdoor probably benefits more from simple PCC or per pixel PCC; or even setAmbientLight( upperHemi, lowerHemi ) that slowly changes colours (lerp) as you move the cameras from region to region.
